sandy/core/cov.py

```diff
@@ -2075,28 +2124,114 @@ def sparse_tables_cholesky(a, rows=1000):
 
     Parameters
     ----------
-    arr : 1d `numpy.ndarray`
-        array with the upper triangular values of the matrix
-    size : `int`
-        dimension of the matrix
+    data_stack : `pd.Dataframe`
+        Stacked dataframe.
+    rows : `int`, optional
+        Number of rows to take into account into each loop. The default
+        is 10000000.
+    index : 1D iterable, optional
+        Index of the final covariance matrix. The default is
+        ["MAT", "MT", "E"].
+    columns : 1D iterable, optional
+        Columns of the final covariance matrix. The default is
+        ["MAT1", "MT1", "E1"].
+    values : `str`, optional
+        Name of the column where the values are located. The default is 'VAL'.
 
     Returns
     -------
-    2d `numpy.ndarray`
-        reconstructed 2d-array with symmetric matrix
+    pivot_matrix : `pd.DataFrame`
+        Covariance matrix created from a stacked data
+
+    Examples
+    --------
+    >>> S = pd.DataFrame(np.array([[1, 1, 1], [0, 2, 1], [0, 0, 1]]))
+    >>> S = S.stack().reset_index().rename(columns = {'level_0': 'dim1', 'level_1': 'dim2', 0: 'cov'})
+    >>> sandy.cov.segmented_pivot_table(S, index=['dim1'], columns=['dim2'], values='cov')
+    dim2	0	1	2
+    dim1
+       0	1	1	1
+       1	0	2	1
+       2	0	0	1
+    """
+    size = data_stack.shape[0]
+    pivot_matrix = pd.DataFrame()
+    for i in range(0, size, rows):
+        partial_pivot = data_stack[i: min(i+rows, size)].pivot_table(
+            index=index,
+            columns=columns,
+            values=values,
+            fill_value=0,
+            aggfunc=np.sum,
+            )
+        pivot_matrix = pd.concat([pivot_matrix, partial_pivot]).fillna(0)           
+    # Because the default axis to concatenate is the 0, some duplicate
+    # index appear with null values. With this groupby, the duplicate axis
+    # disappear, keeping the original values.
+    pivot_matrix = pivot_matrix.groupby(pivot_matrix.index).sum()
+    if len(index) >= 2:
+        # Groupby transform multiindex structure into a tuple. This line
+        # reverse the transformation.
+        pivot_matrix.index = pd.MultiIndex.from_tuples(
+            pivot_matrix.index,
+            name=index,
+            )
+    return pivot_matrix
+
+def triu_matrix(matrix, kind='upper'):
+    """
+    Given the upper or lower triangular matrix , return the full symmetric
+    matrix.
+
+    Parameters
+    ----------
+    matrix : 2d iterable
+        Upper triangular matrix
+    kind : `str`, optional
+        Select if matrix variable is upper or lower triangular matrix. The
+        default is 'upper'
+
+    Returns
+    -------
+    `pd.Dataframe`
+        reconstructed symmetric matrix
+
+    Examples
+    --------
+    >>> S = pd.DataFrame(np.array([[1, 2, 1], [0, 2, 4], [0, 0, 3]]))
+    >>> triu_matrix(S)
+    	0	1	2
+    0	1	2	1
+    1	2	2	4
+    2	1	4	3
+
+    >>> S = pd.DataFrame(np.array([[1, 2, 1], [2, 2, 4], [0, 0, 3]]))

```

Comment:
what's the difference between these two tests? add one line explaining what you are testing